persistent/Database/Persist/TH.hs

```diff
@@ -374,18 +374,18 @@ liftAndFixKeys mps emEntities entityMap unboundEnt =
                             , "\n\nYou can use the References keyword to fix this."
                             ]
                     | otherwise =
-                        zip (fmap (withDbName fieldStore) fieldNames) parentKeyFieldNames
+                        zip (fmap (withDbName fieldStore) fieldNames) (toList parentKeyFieldNames)
                   where
                     parentKeyFieldNames
-                        :: [(FieldNameHS, FieldNameDB)]
+                        :: NonEmpty (FieldNameHS, FieldNameDB)
```

Comment:
just pushing this as far as we can.. even if it's `toList`ed about two lines above.